service/src/test/java/bio/terra/pipelines/db/JobsDaoTest.java

```diff
@@ -0,0 +1,84 @@
+package bio.terra.pipelines.db;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import bio.terra.pipelines.db.exception.DuplicateObjectException;
+import bio.terra.pipelines.service.model.Job;
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class JobsDaoTest extends BaseDaoTest {
+
+  @Autowired JobsDao jobsDao;
+
+  private final String testUserId = "testUser";
+  private final String testPipelineId = "testPipeline";
+
+  private Job createTestJobWithUUID(UUID jobId) {
+    return createTestJobWithUUIDAndUser(jobId, testUserId);
+  }
+
+  private Job createTestJobWithUUIDAndUser(UUID jobId, String userId) {
+    Instant timeSubmitted = Instant.now();
+    String status = "SUBMITTED";
+    return new Job(jobId, userId, testPipelineId, "testVersion", timeSubmitted, null, status);
+  }
+
+  @Test
+  void testWriteValidJob() {
+    String testNewJobId = "deadbeef-dead-beef-aaaa-aaaadeadbeef";
+    UUID jobId = UUID.fromString(testNewJobId);
+    Job newJob = createTestJobWithUUID(jobId);
+    UUID uuidResult = jobsDao.createJob(newJob);
+
+    assertEquals(jobId, uuidResult);
+  }
+
+  @Test
+  void testWriteDuplicateJob() {
+    // repeating the exact same logic as above, except using a job id that we know is preloaded in
+    // the test data
+    String testPresentJobId = "deadbeef-dead-beef-deaf-beefdeadbeef";
+    UUID jobId = UUID.fromString(testPresentJobId);
+    Job newJob = createTestJobWithUUID(jobId);
+
+    assertThrows(DuplicateObjectException.class, () -> jobsDao.createJob(newJob));
+  }
+
+  @Test
+  void testGetCorrectNumberOfRows() {
+    // A test row should exist for this user.
+    List<Job> jobs = jobsDao.getJobs(testUserId, testPipelineId);
+    assertEquals(1, jobs.size());
+
+    // insert another row and verify that it shows up
+    Job newJob = createTestJobWithUUID(UUID.randomUUID());
+
+    jobsDao.createJob(newJob);
+    jobs = jobsDao.getJobs(testUserId, testPipelineId);
+    assertEquals(2, jobs.size());
+  }
+
+  @Test
+  void testCorrectUserIsolation() {
+    String testUserId2 = "testUser2";
+    // A test row should exist for this user.
```

Comment:
super tiny nit: I initially read "this user" as `testUserId2` (the one whose id was just defined in the previous line), but we're really checking the existing job for the existing user (`testUserId`). I'd either change this comment to "A test row should exist for the first user" or move line 67 down a few rows until we're about to call it.

Reply:
good catch.  I rearranged the code and moved the second user id into the top of that method (C-style) instead of at the top of the class, and it left the comment ambiguous.  Rearranged it.